ui/src/LandingPage.js

```diff
@@ -335,6 +374,72 @@ class LandingPage extends Component {
     this.setState({ createPrintTemplateDialogDisplayed: false });
   };
 
+  onCreateSmsTemplate = async () => {
+    var failedValidation = false;
+
+    if (!this.state.packCode.trim()) {
+      this.setState({ packCodeValidationError: true });
+      failedValidation = true;
+    }
+    if (!this.state.notifyTemplateId) {
+      this.setState({
+        notifyTemplateIdValidationError: true,
+      });
+      failedValidation = true;
+    }
+    if (!this.state.template.trim()) {
+      this.setState({ templateValidationError: true });
+      failedValidation = true;
+    } else {
+      try {
+        const parsedJson = JSON.parse(this.state.template);
+        if (!Array.isArray(parsedJson)) {
+          this.setState({ templateValidationError: true });
+          failedValidation = true;
+        } else {
+          const validTemplateItems = ["__uac__", "__qid__"];
```

Comment:
should you leave the validation to the backend so it's not hard-coded into the front-end?

Reply:
Going to leave it for now